mdal/frmts/mdal_ply.cpp

```diff
@@ -418,19 +462,19 @@ void MDAL::DriverPly::addDataset( MDAL::DatasetGroup *group, const std::vector<d
 
   if ( group->dataLocation() == DataOnVertices )
   {
-    assert( values.size() == mesh->verticesCount() );
+    assert( values.size() == mesh->verticesCount()*mult );
   }
 
   if ( group->dataLocation() == DataOnFaces )
   {
-    assert( values.size() == mesh->facesCount() );
+    assert( values.size() == mesh->facesCount()*mult );
     if ( mesh->facesCount() == 0 )
       return;
   }
 
   if ( group->dataLocation() == DataOnEdges )
   {
-    assert( values.size() == mesh->edgesCount() );
+    assert( values.size() == mesh->edgesCount()*mult );
```

Comment:
maybe instead of assert move to the if() condition?

Reply:
Done in the next commit